torchtitan/experiments/flux/train.py

```diff
@@ -92,7 +92,8 @@ def train_step(self, input_dict: dict[str, torch.Tensor], labels: torch.Tensor):
         # the major variables that are used in the training loop.
         model_parts = self.model_parts
         assert len(self.model_parts) == 1
-        model = self.model_parts[0]
+        # explicitely convert flux model to be Bfloat16 no matter FSDP is applied or not
+        model = self.model_parts[0].to(self._dtype)
```

Comment:
I think we should move this to `__init__()`, we don't want to do the conversion every iteration if the model is fixed to an certain dtype.